code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -163,11 +163,22 @@
 	if(recharge_rate && recharge_ticks)
 		to_chat(user, "<span class=\"info\">A small label on a thermocouple notes that it recharges at a rate of [recharge_rate]MJ for every [recharge_ticks<=1?"":"[recharge_ticks] "]oscillator tick[recharge_ticks>1?"s":""].</span>")
 
-/obj/item/weapon/card/emag/attack()
+//only trigger if target is human thus having prosthetic limbs to sabotage
+/obj/item/weapon/card/emag/attack(var/mob/living/target, var/mob/living/attacker)
+	if (!istype(target, /mob/living/carbon/human))
+		return
+	//get target zone with 0% chance of missing
+	var/zone = ran_zone(attacker.zone_sel.selecting, 100)
+	var/datum/organ/external/organ = target.get_organ(zone)
+	target.emag_act(attacker, organ, src)
 	return
 
+//this method gets called from held_item.afterattack as it should, but in the case of somebody with emag clicking on a human we don't want this method to trigger,
+//since the logic for emag attack is in emag/attack
 /obj/item/weapon/card/emag/afterattack(atom/target, mob/user, proximity)
 	var/atom/A = target
+	if (istype(target, /mob/living/carbon/human)) 
```

Comment:
iirc you don't need this exception if you make attack return TRUE, as this will end the attack

Reply:
Just tried it - doesn't work. Maybe it's for other attack methods? 
Here is the code for item attack:
https://github.com/vgstation-coders/vgstation13/blob/1fd2959db8d791e485716848d333bde73f1db6b0/code/_onclick/item_attack.dm#L56-L63

Reply:
There is this check on preattack, maybe you meant that?

https://github.com/vgstation-coders/vgstation13/blob/84191a1fc117f0dde70d8cbd65f11512326f55b2/code/_onclick/click.dm#L129-L150

Reply:
why not make the emag a preattack?

Reply:
That would also mean moving the logic to emag objects also in preattack? I can't just have the human emagging logic in preattack and the other logic in afterattack. And even if I move all logic to preattack, I am unsure if that would break everything else - like other methods checking for proximity leaving fingerprints etc?

Reply:
Isn't that why the method is called preattack in the first place? To check all that stuff, and the actual attack logic should be in the attack method.

Reply:
However I can move everything to afterattack it seems.

Reply:
There, I even left some handy comments, so nobody gets confused as we did.